src/todo/todo.service.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Todo } from 'src/entities/todo.entity';
+import { User } from 'src/entities/user.entity';
+import { CreateTodoDto } from './dto/create-todo.dto';
+import { TodoState } from 'src/entities/todo.entity';
+import { plainToClass } from 'class-transformer';
+import { UpdateTodoDto } from './dto/update-todo.dto';
+
+@Injectable()
+export class TodoService {
+  constructor(
+    @InjectRepository(Todo) private todoRepository: Repository<Todo>,
+    @InjectRepository(User) private userRepository: Repository<User>,
+  ) {}
+
+  async create(createTodoPayload: CreateTodoDto): Promise<Todo> {
+    try {
+      const user = await this.userRepository.findOne({
+        where: { id: createTodoPayload.userId },
+      });
+
+      if (!user) {
+        throw new NotFoundException(
+          `User with ID ${createTodoPayload.userId} not found`,
+        );
+      }
+
+      const state = createTodoPayload.state ?? TodoState.PENDING;
+
+      const todo = this.todoRepository.create({
+        title: createTodoPayload.title,
+        description: createTodoPayload.description,
+        state: state,
+        userId: createTodoPayload.userId,
+        user: user,
+      });
+
+      const savedTodo = await this.todoRepository.save(todo);
+
+      const sanitizedTodo = plainToClass(Todo, savedTodo, {
+        excludeExtraneousValues: true,
+      });
```

Comment:
If the response is some other DTO, then you'll need to do this twice. So, find another alternative. We shouldn't need to sanitize the repository response.